### ISSUE TITLE
[$250] IOU - Web - Loading spinner appears under expense every time open the expense page

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#45227]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.6-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Go to FAB> Submit expense> Complete the flow
2. Go to thread transaction page
3. Go back to request page ( switch chat if needed)

## Expected Result:
There should be no loading spinner when open the request page

## Actual Result:
Loading spinner appears under expense every time open the request page

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/475197b5-2e94-40bd-b3c6-fcd274efe1eb

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~018c260c51c3955b65</li>
        <li>Upwork Job ID: 1813342162639120406</li>
        <li>Last Price Increase: 2024-07-16</li>
<li>Automatic offers: </li>
<ul>
<li>paultsimura | Reviewer | 103347808</li>
<li>ishpaul777 | Contributor | 103347809</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @ishpaul777</details>

**Comment by lanitochka17** [comment]
@mallenexpensify  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by lanitochka17** [comment]
We think that this bug might be related to #vip-vsp

**Comment by bernhardoj** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Loading indicator shows when we go back from a thread of the last message of a chat.

### What is the root cause of that problem?
When we open a thread of the last message of a chat, then go back usign the parent link (on the header), a ReadNewestAction request is called which will show the loading indicator.
https://github.com/Expensify/App/blob/978b833cf10aa22f3f20257f645e31a2f64492b3/src/pages/home/report/ListBoundaryLoader.tsx#L99-L109

This happens after https://github.com/Expensify/App/pull/39454. The read newest action request is triggered from this block.
https://github.com/Expensify/App/blob/978b833cf10aa22f3f20257f645e31a2f64492b3/src/pages/home/report/ReportActionsView.tsx#L431-L433
https://github.com/Expensify/App/blob/978b833cf10aa22f3f20257f645e31a2f64492b3/src/pages/home/report/ReportActionsView.tsx#L297-L308

Before the linked PR, we have a `hasNewestReportAction` in the condition. `hasNewestReportAction` tells us whether the last report action in the list is the newest one or not. If it's the newest one, then we don't request for read newest action. But the PR removed it because of this [issue](https://github.com/Expensify/App/pull/30269#discussion_r1547981656), so the read newest action request is always triggered.

### What changes do you think we should make in order to solve the problem?

The solution is to only request for read newest action if the `hasNewestReportAction` is false in here.
https://github.com/Expensify/App/blob/978b833cf10aa22f3f20257f645e31a2f64492b3/src/pages/home/report/ReportActionsView.tsx#L300-L303

**Comment by bernhardoj** [comment]
Added more explanation to the proposal

**Comment by mallenexpensify** [comment]
@bernhardoj is this the loading spinner that's referenced in the OP?
<img width="500" alt="image" src="https://github.com/user-attachments/assets/c79ac0fe-9ea2-44da-9d22-bb47c03353e3">

**Comment by bernhardoj** [comment]
I believe it's this one, the read new action loading indicator.

<img width="1426" alt="image" src="https://github.com/user-attachments/assets/1047bb9c-b0dc-4a8b-9f55-dc14097a9c97">

**Comment by mallenexpensify** [comment]
@paultsimura can you attempt reproduction?  If you're able to reproduce and the steps in the OP need to be updated, please outline them and I'll update the OP.  I tried to reproduce yesterday and today with no luck (I think I might not have been holding it right).  thx

**Comment by mallenexpensify** [comment]
If it's a reproducible bug and we can fix it, we likely should.  That said... might this be occurring anywhere else in the app? Or.. would the fix here fix/improve anything else?

**Comment by bernhardoj** [comment]
It will fix https://github.com/Expensify/App/issues/45387 too, which is easier to reproduce.

**Comment by mallenexpensify** [comment]
Thanks @bernhardoj, @paultsimura plz proceed with reviewing @bernhardoj 's proposal above

**Comment by mallenexpensify** [comment]
@trjExpensify , added to Wave Collect cuz the bug occurs on the expense page. It's def polish and/or low priority.

**Comment by trjExpensify** [comment]
We should for sure get eyes on [this RCA and proposal](https://github.com/Expensify/App/issues/45227#issuecomment-2222637243) from the crew that worked on this comment linking initiative, as it was a complex project. So, CC'ing @puneetlath @jasperhuangg @roryabraham @rayane-djouah @rushatgabhane as a few peeps mentioned in the linked issues I saw above.

**Comment by janicduplessis** [comment]
@ishpaul777 I think so too, I will have a look at the proposal soon.

**Comment by bernhardoj** [comment]
PR is ready

cc: @ishpaul777

**Comment by mallenexpensify** [comment]
Contributor: @ owed $250 via NewDot 
Contributor+: @ishpaul777 paid $250 via Upwork 

Thx @ishpaul777, can you fill out the BZ checklist plz? 

**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@] Determine if we should create a regression test for this bug.
- [ ] [@] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@BZ] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:
- [ ]

**Comment by bernhardoj** [comment]
Requested in ND.

**Comment by mallenexpensify** [comment]
Forgot to tag @bernhardoj in the payment summary, updated. 

@ishpaul777 BZ checklist plzzzzzzzz

**Comment by mallenexpensify** [comment]
Thanks @ishpaul777 , closing, test case is here
- https://github.com/Expensify/Expensify/issues/424780

**Comment by JmillsExpensify** [comment]
$250 approved for @bernhardoj
